### PR TITLE
Improv cmake

### PR DESCRIPTION
**Pull request by 9prady9**

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -24,7 +24,8 @@ if (NOT TARGET gtest)
   add_subdirectory(googletest/googletest EXCLUDE_FROM_ALL)
```
Comment (acrlakshman) [inline test/CMakeLists.txt]: Can we be able to replace this with ExternalProjects_Add(...). I started experimenting this feature this week, but haven't succeeded yet, just putting that thought here.

Comment (9prady9) [reply test/CMakeLists.txt]: You are welcome to try it out. There isn't any specific advantage of that over the submodule I think

Comment (acrlakshman) [reply test/CMakeLists.txt]: Oh, thanks for the info. I just came across, just wanted to hear your comments. Looks like it may not be advantageous then.

Comment (9prady9) [reply test/CMakeLists.txt]: meshio is header only and gtest is for tests alone, so it's okay I guess

Comment (acrlakshman) [reply test/CMakeLists.txt]: Sure, that' ok, I asked this to more like I wanted to learn new features in cmake...

**Pull request merged**